packages/app/spammer/spammer.go

```diff
@@ -83,9 +83,14 @@ func (s *Spammer) run(rate int, payloadSize uint64, timeUnit time.Duration, imif
 		case <-s.shutdown:
 			return
 		case <-ticker.C:
-			estimatedDuration := s.estimateFunc()
 			// TODO: only sleep if estimate > some threshold.
-			time.Sleep(estimatedDuration)
+			for {
+				if estimatedDuration := s.estimateFunc(); estimatedDuration > 0 {
+					time.Sleep(estimatedDuration)
+				} else {
+					break
+				}
+			}
```

Comment:
Would the following be cleaner?

```suggestion
			for estimatedDuration := s.estimateFunc(); estimatedDuration > 0; estimatedDuration = s.estimateFunc() {
				time.Sleep(estimatedDuration)
			}
```